tests/unit/s2n_security_policy_cert_preferences_test.c

```diff
@@ -128,11 +128,15 @@ int main(int argc, char **argv)
             EXPECT_OK(s2n_security_policy_validate_certificate_chain(&test_sp, &chain));
         };
 
-        /* an invalid root signature is ignored */
+        /* an invalid root signature causes a failure */
         {
+            struct s2n_cert_info valid_root = root.info;
             root.info.signature_nid = invalid_sig_nid;
             root.info.signature_digest_nid = invalid_hash_nide;
-            EXPECT_OK(s2n_security_policy_validate_certificate_chain(&test_sp, &chain));
+            EXPECT_ERROR_WITH_ERRNO(s2n_security_policy_validate_certificate_chain(&test_sp, &chain), 
+                    S2N_ERR_SECURITY_POLICY_INCOMPATIBLE_CERT);
+            /* restore root values */
+            root.info = valid_root;
```

Comment:
Nit: This is probably fine, but edging towards dangerous. Similar to the previous comment: tests should share as little state as possible, and what state they do share should be const if at all possible.